warehouse/query-core/src/main/java/datawave/query/jexl/functions/GroupingRequiredFilterFunctions.java

```diff
@@ -224,6 +225,12 @@ public static Collection<?> matchesInGroup(Object... args) {
      */
     private static void manageMatchesInGroupRemainingArgs(Object fieldValue, String regex, String context, Collection<ValueTuple> allMatches,
                     ValueTuple currentMatch, int positionFromRight) {
+        Key candidateMetadata = ValueTuple.getSourceMetadata(fieldValue);
+        Key currentMetadata = ValueTuple.getSourceMetadata(currentMatch);
+        if (candidateMetadata != null && currentMetadata != null && !currentMetadata.equals(candidateMetadata)) {
```

Comment:
So if they are both null, then we cannot match?  I guess in general they should never be null but ok....